src/Microsoft.AspNet.Server.Kestrel/Filter/LibuvStream.cs

```diff
@@ -85,7 +85,7 @@ public override void Flush()
 
         private Task<int> ReadAsync(ArraySegment<byte> buffer)
         {
-            return _input.ReadAsync(buffer);
+            return _input.ReadAsync(buffer.Array, buffer.Offset, buffer.Count);
```

Comment:
Worth just overloading this on `SocketInput`? Notice a bunch of other methods here create an array segment to call this method.

At least least, you could clean all of those up to avoid creating the `ArraySegment`